portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,93 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
+  private List<String> commentArray = new ArrayList<>();
+
+  /** Converts Comment ArrayList to Json */
+  public String toJson() throws IOException {
+   return new Gson().toJson(commentArray);
+  }
+
+  /** Redirection to the main page */
+  private void doRedirect(HttpServletResponse response) throws IOException {
+    response.sendRedirect("/index.html");
+  }
+
+  /** Write a new message to DataStore */
+  private void toDatastore(String comment) throws IOException {
+    // Create a new entity to save in datastore 
+    Entity newComment = new Entity("Comment");
+
+    // Set the entity's values { key: value } 
+    newComment.setProperty("comment", comment); 
+
+    // Call to get datastore service
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(newComment); // Write the entity into datastore 
+  }
+
+  private void dataServletResponse(HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    response.getWriter().println(toJson());
+  }
+
+  public void loadComments() throws IOException {
+    Query commentsQuery = new Query("Comment"); // Get previous stored comments
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    // Prepare ton instance the past stored comments
+    PreparedQuery comments = datastore.prepare(commentsQuery);
+
+    for(Entity commentEntity : comments.asIterable()) {
+      // Get the value of every stored comment
+      String comment = (String) commentEntity.getProperty("comment");
+      commentArray.add(comment); // Add the value to the comments array
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String value = request.getParameter("user-comment");
+   
+    /** Redirection if value is empty or accidental click */
+    if(value == null || value.length() == 0) {
+      doRedirect(response);
+      return;
+    } else {
+      commentArray.add(value); // Add every new submition to recorded comments 
+      toJson();
+      toDatastore(value);
+    }
```

Comment:
```suggestion
    }
    commentArray.add(value); // Add every new submition to recorded comments 
    toJson();
    toDatastore(value);
```
Since you return early from the function, I think you can remove the else case